tests/datastreams/test_datastreams.py

```diff
@@ -144,3 +146,154 @@ def test_piping_readers(app, zip_file, json_element):
+                                <subfield code="a">Stahlbetonbauwerke mit großer Spannweite, eingesetzt im Industriebau, z.b.Paketposthalle München; teilweise heute unter Denkmalschutz</subfield>
+                            </datafield>
+                        </record>
+                    </metadata>
+                </record>
+            </ ListRecords>
+        </OAI-PMH>
+    """
+    httpserver.expect_request("/oai/repository").respond_with_data(
+        response_data=response_data, mimetype="application/xml"
+    )
+    ds_config = {
+        "readers": [
+            {
+                "type": "oai-pmh",
+                "args": {
+                    "base_url": httpserver.url_for("/oai/repository"),
+                    "metadata_prefix": "MARC21plus-1-xml",
+                    "set": "authorities:sachbegriff",
+                    "from_date": "2024-01-01T09:00:00Z",
+                    "until_date": "2024-01-31T10:00:00Z",
+                },
+            },
+        ],
+        "writers": [{"type": "test"}],
+    }
+
+    datastream = DataStreamFactory.create(
+        readers_config=ds_config["readers"],
+        writers_config=ds_config["writers"],
+    )
+    iter = datastream.process()
+    for count, entry in enumerate(iter, start=1):
+        assert "record" in entry.entry
+        break
+
+
+def test_oaipmh_reader_no_records_match(app, httpserver):
+    response_data = """
+        <OAI-PMH xsi:schemaLocation="http://www.openarchives.org/OAI/2.0/ http://www.openarchives.org/OAI/2.0/OAI-PMH.xsd" xmlns="http://www.openarchives.org/OAI/2.0/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+            <responseDate>2024-05-29T13:09:44Z</responseDate>
+            <request metadataPrefix="MARC21plus-1-xml" set="authorities:sachbegriff" verb="ListRecords" from="2024-01-01T09:00:00Z" until="2024-01-01T10:00:00Z">https://services.dnb.de/oai/repository</request>
+            <error code="noRecordsMatch"/>
+        </OAI-PMH>
+    """
+    httpserver.expect_request("/oai/repository").respond_with_data(
+        response_data=response_data, mimetype="application/xml"
+    )
+    ds_config = {
+        "readers": [
+            {
+                "type": "oai-pmh",
+                "args": {
+                    "base_url": httpserver.url_for("/oai/repository"),
+                    "metadata_prefix": "MARC21plus-1-xml",
+                    "set": "authorities:sachbegriff",
+                    "from_date": "2024-01-01T09:00:00Z",
+                    "until_date": "2024-01-01T10:00:00Z",
+                },
+            },
+        ],
+        "writers": [{"type": "test"}],
+    }
+
+    datastream = DataStreamFactory.create(
+        readers_config=ds_config["readers"],
+        writers_config=ds_config["writers"],
+    )
+    stream_iter = datastream.process()
+    try:
+        _ = next(stream_iter)
+        assert False
+    except ReaderError:
+        assert True
```

Comment:
```suggestion
    with pytest.raises(ReaderError):
        next(stream_iter)
```

This is what I used in a few tests called `test_ror_http_reader_...`, I think it should work.